File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt the user for input
    string text = get_string("Text: ");

    // Initilize counters for letters, words, and sentances
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Count letters
        if (isalpha(text[i]))
        {
            letters++;
        }
        // Count words by counting spaces
        else if (isspace(text[i]))
        {
            words++;
        }
        // Count sentences by detecting end of sentence puntuation
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    // Calculate average number of letters per 100 words (L) and average number of sentences per 100 words (S)
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    // Calculate the Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Determine and print the grade level based on the index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
import cs50

# Function to calculate Coleman-Liau index
def calculate_index(letters, words, sentences):
    L = (letters / words) * 100
    S = (sentences / words) * 100
    return round(0.0588 * L - 0.296 * S - 15.8)

# Prompt the user for input
text = cs50.get_string("Text: ")

# Initialize counters for letters, words, and sentences
letters = words = sentences = 0

# Loop through each character in the input text
for char in text:
    # Count letters
    if char.isalpha():
        letters += 1
    # Count words by counting spaces
    elif char.isspace():
        words += 1
    # Count sentences by detecting end of sentence punctuation
    elif char in ['.', '?', '!']:
        sentences += 1

# Add one to words for the last word
words += 1

# Calculate the Coleman-Liau index
index = calculate_index(letters, words, sentences)

# Determine and print the grade level based on the index
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
